conda_smithy/schema/models.py

```diff
@@ -0,0 +1,476 @@
+        ),
+    ] = None
+
+
+class ChannelPriorityConfig(str, Enum):
+    STRICT = "strict"
+    FLEXIBLE = "flexible"
+    DISABLED = "disabled"
+
+
+class CondaForgeChannels(BaseModel):
+    """This represents the channels to grab packages from during builds and which channels/labels to push to on anaconda.org after a package has been built. The channels variable is a mapping with sources and targets."""
+
+    sources: List[str] = Field(
+        default=["conda-forge"],
+        description="sources selects the channels to pull packages from, in order",
+    )
+
+    targets: List[List[str]] = Field(
+        default=[["conda-forge", "main"]],
+        description="targets is a list of 2-lists, where the first element is the channel to push to and the second element is the label on that channel",
+    )
+
+
+class CondaBuildConfig(BaseModel):
+    pkg_format: Union[int, str, None] = Field(
+        description="The package format for conda build. This can be one of 1, 2, or tar. The default is 2.",
+        default="2",
+    )
+
+    zstd_compression_level: int = Field(
+        default=16,
+        description="The compression level for the zstd compression algorithm for .conda artifacts. conda-forge uses a default value of 16 since its artifacts can be large.",
+    )
+
+    @field_validator("pkg_format")
+    def pkg_format_validator(cls, v):
+        if not v:
+            return v
+
+        valid_values = ["2", None]
+        if v not in valid_values:
+            raise ValueError(
+                f"Invalid value {v} for pkg_format. Valid values are {valid_values}"
+            )
+        return v
+
+
+class CondaForgeDocker(BaseModel):
+    model_config = ConfigDict(extra="allow")
+
+    executable: str = Field(
+        description="The executable for Docker", default="docker"
+    )
+
+    fallback_image: str = Field(
+        description="The fallback image for Docker",
+        default="quay.io/condaforge/linux-anvil-comp7",
+    )
+
+    command: str = Field(
+        description="The command to run in Docker", default="bash"
+    )
+
+    interactive: bool = Field(
+        description="Whether to run Docker in interactive mode", default=None
+    )
+
+    image: str = Field(description="The image for Docker", default=None)
+
+
+class ShellCheck(BaseModel):
+    enabled: bool = Field(
+        description="Whether to use shellcheck to lint shell scripts",
+        default=False,
+    )
+
+
+class ConfigModel(BaseModel):
+    appveyor: Dict[str, Any] = Field(
+        default={"image": "Visual Studio 2017"},
+        description="AppVeyor CI settings. This is usually read-only and should not normally be manually modified. Tools like conda-smithy may modify this, as needed.",
+    )
+
+    azure: AzureConfig = AzureConfig()
+
+    bot: BotConfig = BotConfig()
+
+    build_platform: Annotated[
+        Union[Dict[Platforms, Platforms], None],
+        Field(
```

Comment:
Why use this as `Annotated` instead of `Field` in value (as done elsewhere)?

Reply:
Oh, my bad. Sometimes I like to use Anotated everywhere, and sometimes I use Field only. Though in this case, I started with Fields, then moved everything to Annotaded (I guess the reason was to be more compact, and found it more elegant); though as I was testing and changing the model, for quickness, I used Fields to test faster

Reply:
Update all to Fields :smile: